org.abapgit.adt.ui/src/org/abapgit/adt/ui/internal/repositories/actions/SwitchbranchAction.java

```diff
@@ -0,0 +1,84 @@
+package org.abapgit.adt.ui.internal.repositories.actions;
+
+import org.abapgit.adt.backend.model.abapgitrepositories.IRepository;
+import org.abapgit.adt.ui.AbapGitUIPlugin;
+import org.abapgit.adt.ui.internal.i18n.Messages;
+import org.abapgit.adt.ui.internal.repositories.AbapGitView;
+import org.abapgit.adt.ui.internal.repositories.exceptions.PackageRefNotFoundException;
+import org.abapgit.adt.ui.internal.wizards.AbapGitWizardSwitchBranch;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.jface.action.Action;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.jface.wizard.WizardDialog;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.swt.widgets.Shell;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.plugin.AbstractUIPlugin;
+
+import com.sap.adt.tools.core.project.AdtProjectServiceFactory;
+import com.sap.adt.tools.core.ui.packages.AdtPackageServiceUIFactory;
+import com.sap.adt.tools.core.ui.packages.IAdtPackageServiceUI;
+
+public class SwitchbranchAction extends Action {
+
+	private IRepository selRepo;
+	private final AbapGitView AbapGitView;
+	private IProject project;
+
+	public SwitchbranchAction(AbapGitView view) {
+		super(Messages.AbapGitView_action_switch_branch);
+		setToolTipText(Messages.AbapGitView_action_switch_branch);
+		setImageDescriptor(AbstractUIPlugin.imageDescriptorFromPlugin(AbapGitUIPlugin.PLUGIN_ID, "/icons/etool/branches_obj.png")); //$NON-NLS-1$
+		this.AbapGitView = view;
+	}
+
+	@Override
+	public void run() {
+		this.project = getProject();
+		this.selRepo = getRepository();
+		if (this.selRepo != null) {
+			String destination = AdtProjectServiceFactory.createProjectService().getDestinationId(this.project);
+			if (getPackageAndRepoType(destination)) {
+				WizardDialog dialog = new WizardDialog(this.AbapGitView.getViewSite().getShell(),
+						new AbapGitWizardSwitchBranch(this.project, this.selRepo, destination));
+				dialog.open();
+			}
```

Comment:
```suggestion
			try {
					WizardDialog dialog = new WizardDialog(this.AbapGitView.getViewSite().getShell(),
							new AbapGitWizardSwitchBranch(this.project, this.selRepo, destination));
					dialog.open();
				} catch (PackageRefNotFoundException | ResourceException e) {
					MessageDialog.openError(this.AbapGitView.getViewSite().getShell(), "Switch Branch Wizard Error",
							e.getLocalizedMessage());
				}
```

- The idea is that the repo type and package info is retrieved while creating the object of the wizard
- This wizard also sends the new exception for package ref not found
- Also, the post call to backend can possibly throw a ResourceException, in case there is an issue in backend. This can be explicitly handled here.
- The shell to we can get from the abapgit view